### ISSUE TITLE
Multithread search not display all Pokemon

### ISSUE THREAD

**Issue by slonce** [AHAAAAAAA/PokemonGo-Map#1477]
Good afternoon. I did not understand why, when the use does not show a lot of Pokemon? Although the area scanned? It is necessary to use MySql? Use Python 2.7 and last dev

In 1st i m use Multithread search 10 (start runserver.py -a google -u **** -p **** -l "50.255022, 28.658587" -st 25 -H 0.0.0.0 -P 3141 -t 20 -sd 10), in second start Launched in different regions in 1 thread

![1](https://cloud.githubusercontent.com/assets/10965693/17077274/a171eb96-50d2-11e6-931c-43bf5d95c49c.jpg)
![2](https://cloud.githubusercontent.com/assets/10965693/17077273/a1703e72-50d2-11e6-9b0c-fab3e4856e12.jpg)

**Comment by denniskai1234** [comment]
Where did you find something with MySQL?

**Comment by Fruktz** [comment]
I have noticed this too. This makes searches so much slower

**Comment by cNyhuus** [comment]
There is alot of talking about this problem in https://github.com/AHAAAAAAA/PokemonGo-Map/issues/1463

**Comment by slonce** [comment]
Yes, I see, but did not solve the problem. In one stream Pokemon much more than in the same 5 or 15

**Comment by DieNot** [comment]
Go to pogom/search.py and replace the `def search_thread(args)` with this:
```
def search_thread(args):
    i, total_steps, step_location, step, sem = args

    log.info('Scanning step {:d} of {:d} started.'.format(step, total_steps))
    log.debug('Scan location is {:f}, {:f}'.format(step_location[0], step_location[1]))

    response_dict = {}
    failed_consecutive = 0
    while not response_dict:
		sem.acquire()
		response_dict = send_map_request(api, step_location) 
		sem.release()
                if response_dict: 
                    try:
                        parse_map(response_dict, i, step, step_location) 
		    except KeyError:
                        log.error('Scan step {:d} failed. Response dictionary key error.'.format(step))
                        failed_consecutive += 1
                        if(failed_consecutive >= config['REQ_MAX_FAILED']):
                            log.error('Niantic servers under heavy load. Waiting before trying again')
                            time.sleep(config['REQ_HEAVY_SLEEP'])
                            failed_consecutive = 0
                else:
                     log.info('Map Download failed. Trying again.')

    time.sleep(config['REQ_SLEEP'])
```
Now multithreading should work fine.

**Comment by mcmanus82** [comment]
@DieNot Besides some indent issues, it seems to fix the problem. :+1:

**Comment by marauder37** [comment]
Well I'll test anything once, but I don't see how moving the locking to a non-critical section actually helps. Did you test on Windows?

**Comment by marauder37** [comment]
wcastello points out that locking send_map_request is basically the same as running --threads 1

**Comment by wcastello** [comment]
I don't think this is a solution to the real problem. Like I was talking to @marauder37 send_map_request() is the slowest call here, this just makes every thread run much slower and and that's why you don't get locks. 
I still don't understand why the lock around the parse_map() wouldn't work though since parse_map is the one calling the InsertQuery()'s. If anyone can explain that I'd like to know.
